packages/app-project/src/components/AuthModal/shared/components/withCustomFormik/withCustomFormik.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Formik } from 'formik'
+
+function withCustomFormik (WrappedComponent) {
+  class FormikHOC extends React.Component {
+    constructor () {
+      super()
+
+      this.handleBlurWithCallback = this.handleBlurWithCallback.bind(this)
+      this.handleChangeWithCallback = this.handleChangeWithCallback.bind(this)
+    }
+
+    handleBlurWithCallback (event, formikProps) {
+      formikProps.handleBlur(event)
+      this.props.onBlur(event, formikProps)
+    }
+
+    handleChangeWithCallback (event, formikProps) {
+      formikProps.handleChange(event)
+      this.props.onChange(event, formikProps)
+    }
+
+    render () {
+      const { initialValues, onSubmit, ...rest } = this.props
+      return (
+        <Formik
+          initialValues={initialValues}
+          onSubmit={onSubmit}
+          {...rest}
+        >
+          {(innerProps) => {
+            const {
+              handleBlur,
+              handleChange,
+              handleReset,
+              handleSubmit,
+              ...restOfInnerProps
+            } = innerProps
+
+            return (
+              <WrappedComponent
+                handleBlur={(event) => this.handleBlurWithCallback(event, innerProps)}
+                handleChange={(event) => this.handleChangeWithCallback(event, innerProps)}
+                {...restOfInnerProps}
+              />
```

Comment:
`handleSubmit` and `handleReset` are missing here.

Reply:
They're being passed down with the `restOfInnerProps`. Formik supports `onSubmit` and `onReset` custom handlers, so I don't really have to do the hack like I have for change and blur. I had them explicitly defined for the props passed down here in the previous commit because I wanted the prop namings to be consistent, however, since I reverted back to `handleChange`, `handleBlur`, etc, I don't really need to do that again.

Reply:
You’ve declared them separately on the previous line. This is where it would be useful to have tests for the props that are passed down to the wrapped form.

Reply:
Yep I did. I'll clean this up and look into adding forwarded refs

Reply:
If you want to get this out of the way quickly, I think `wrapper.dive().find('MockForm').props()` should also work for getting a reference to the form and testing that the handlers have been added properly.

Then ref forwarding could be another PR. Depends how much more work it adds.

Reply:
I've added the forwarded refs and refactored the tests for the onChange and onBlur handlers. The onChange handler is working, however, I could not get the onBlur one to work at all. I even manually tested to see if the original formik `handleBlur` was working and it does not appear to be firing at all. I have no idea why and I'd like to investigate this separately with the LoginForm in a follow up because I know the blur event handlers are working there with the direct use of the Formik API. For now, I removed the customized onBlur handling.

Reply:
Changes look good to me. I’ll pull it down tomorrow and test it locally. The only blocking issue today was that submission had broken, and it looks like that’s fixed now.